components/brave_wallet_ui/common/hooks/use_is_account_syncing.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import getAPIProxy from '../async/bridge'
+
+// Types
+import { BraveWallet } from '../../constants/types'
+
+// Selectors
+import { useSafeWalletSelector } from './use-safe-selector'
+import { WalletSelectors } from '../selectors'
+
+// Hooks
+import { useGetIsSyncInProgressQuery } from '../slices/api.slice'
+
+export const useIsAccountSyncing = (accountId?: BraveWallet.AccountId) => {
+  // State
+  const [isSyncing, setIsSyncing] = React.useState<
+    'started' | 'stopped' | undefined
+  >()
+
+  // Selectors
+  const isZCashShieldedTransactionsEnabled = useSafeWalletSelector(
+    WalletSelectors.isZCashShieldedTransactionsEnabled
+  )
+
+  // Queries
+  const { data: isSyncAlreadyInProgress } = useGetIsSyncInProgressQuery(
+    isZCashShieldedTransactionsEnabled &&
+      !isSyncing &&
+      accountId &&
+      accountId.coin === BraveWallet.CoinType.ZEC
+      ? accountId
+      : skipToken
+  )
+
+  // Effects
+  React.useEffect(() => {
+    const zcashWalletServiceObserver =
+      new BraveWallet.ZCashWalletServiceObserverReceiver({
+        onSyncStart: (id: BraveWallet.AccountId) => {
+          if (accountId && accountId.uniqueKey === id.uniqueKey) {
+            setIsSyncing('started')
```

Comment:
Should we set an account Id to `setIsSyncing` state, so that we don't disable transactions for other ZCash accounts that may not be syncing?